### ISSUE TITLE
Improve R code documentation

### ISSUE THREAD

**Issue by IshmaelBelghazi** [IshmaelBelghazi/bigoptim#5]
Improve R code Documentation.

**Comment by tdhock** [comment]
The most important thing for the documentation is to write some examples of how to compute the model and predictions using your code. also some speed and accuracy comparisons with glmnet would be nice.

**Comment by IshmaelBelghazi** [comment]
Agreed. I'll add the examples this week and the benchmarks as soon as I have added the remaining glm cost functions and gradients.
